metrics-exporter/src/bin/io_engine/collector/pool_stat.rs

```diff
@@ -0,0 +1,185 @@
+use super::init_gauge_vec;
+use crate::{cache::Cache, client::pool_stat::PoolIoStat, get_node_name};
+use prometheus::{
+    core::{Collector, Desc},
+    GaugeVec,
+};
+use std::{fmt::Debug, ops::Deref};
+use tracing::error;
+
+/// Collects Pool IoStat metrics from cache.
+#[derive(Clone, Debug)]
+pub(crate) struct PoolIoStatsCollector {
+    pool_bytes_read: GaugeVec,
+    pool_num_read_ops: GaugeVec,
+    pool_bytes_written: GaugeVec,
+    pool_num_write_ops: GaugeVec,
+    pool_read_latency_us: GaugeVec,
+    pool_write_latency_us: GaugeVec,
+    descs: Vec<Desc>,
+}
+
+impl Default for PoolIoStatsCollector {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+/// Initialize all the metrics to be defined for pools iostat collector.
+impl PoolIoStatsCollector {
+    /// Initialize all the metrics to be defined for pools iostat collector.
+    pub fn new() -> Self {
+        let mut descs = Vec::new();
+
+        let pool_bytes_read =
+            init_gauge_vec("bytes_read", "Total bytes read on the pool", &mut descs);
+        let pool_num_read_ops = init_gauge_vec(
+            "num_read_ops",
+            "Number of read operations on the pool",
+            &mut descs,
+        );
+        let pool_bytes_written = init_gauge_vec(
+            "bytes_written",
+            "Total bytes written on the pool",
+            &mut descs,
+        );
+        let pool_num_write_ops = init_gauge_vec(
+            "num_write_ops",
+            "Number of write operations on the pool",
+            &mut descs,
+        );
+        let pool_read_latency_us = init_gauge_vec(
+            "read_latency_us",
+            "Total read latency on the pool in usec",
+            &mut descs,
+        );
+        let pool_write_latency_us = init_gauge_vec(
+            "write_latency_us",
+            "Total write latency on the pool in usec",
+            &mut descs,
+        );
+
+        Self {
+            pool_bytes_read,
+            pool_num_read_ops,
+            pool_bytes_written,
+            pool_num_write_ops,
+            pool_read_latency_us,
+            pool_write_latency_us,
+            descs,
+        }
+    }
+}
+
+impl Collector for PoolIoStatsCollector {
+    fn desc(&self) -> Vec<&Desc> {
+        self.descs.iter().collect()
+    }
+
+    fn collect(&self) -> Vec<prometheus::proto::MetricFamily> {
+        let c = match Cache::get_cache().lock() {
+            Ok(c) => c,
+            Err(error) => {
+                error!(%error,"Error while getting cache resource");
+                return Vec::new();
+            }
+        };
+        let cp = c.deref();
+        let mut metric_family = Vec::with_capacity(6 * cp.pool_iostat().pool_stats.capacity());
```

Comment:
What is this 6 for, 6 counters? Can we do this without hardcoding the 6?

```suggestion
        let mut metric_family = Vec::with_capacity(6 * cp.pool_iostat().pool_stats.capacity());
```

Reply:
Yes its supposed to be 6 counters